lib/fxpay/products.js

```diff
@@ -8,38 +8,66 @@
   var utils = fxpay.getattr('utils');
   var API = fxpay.getattr('api').API;
 
-  exports.all = function products_all(callback) {
-    var promise = new Promise(function(resolve, reject) {
-      var allProducts = [];
 
-      var api = new API(settings.apiUrlBase);
-      var origin = utils.getSelfOrigin();
-      if (!origin) {
-        return reject(errors.InvalidApp('an origin is needed to get products'));
-      }
-      origin = encodeURIComponent(origin);
-      var url;
-
-      if (settings.fakeProducts) {
-        settings.log.warn('about to fetch fake products');
-        url = '/payments/stub-in-app-products/';
-      } else {
-        settings.log.info('about to fetch real products for app',
-                          origin);
-        url = '/payments/' + origin + '/in-app/?active=1';
-      }
+  exports.get = function products_get(productId, opt) {
+    return buildProductReceiptMap().then(function() {
+      return new Promise(function(resolve, reject) {
+        exports.getById(productId, function(error, product) {
+          if (error) {
+            settings.log.error(
+              'no existing product with productId=' + productId +
+              '; error: ' + error);
+            return reject(error);
+          }
+
+          resolve(product);
+        }, opt);
+      });
+    });
+  };
+
 
-      api.get(url, function(err, result) {
-        if (err) {
-          return reject(err);
+  exports.all = function products_all(callback) {
```

Comment:
yeah, true. It was useful for adb logcat but now we have access to WebIDE for devices which fixes things